src/gebt_poc/field_data.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+#include "mesh.h"
+
+namespace openturbine::gebt_poc {
+
+enum class Field {
+    Coordinates,
+    Velocity,
+    Acceleration,
+    AlgorithmicAcceleration,
+    CoordinatesNext,
+    AlgorithmicAccelerationNext,
+    DeltaCoordinates,
+    Weight,
+    StiffnessMatrix
+};
+
+class FieldData {
+public:
+    FieldData(const Mesh& mesh, int number_of_quadrature_points) {
+        auto number_of_nodes = mesh.GetNumberOfNodes();
+        auto number_of_elements = mesh.GetNumberOfElements();
+
+        coordinates_ = Kokkos::View<double**>("coordinates", number_of_nodes, 3);
```

Comment:
Perhaps.  I believe that Kokkos defaults to the layout (left vs right) such that you usually want to be parallel over the left most dimension.  I.E. `parallel_for(num_nodes, ...);`  is probably the most performant pattern: on a host build, the spatial-dim components will be contiguous for caching, while on a device they will be scattered on device for proper coalescing.  

That being said, once we have decent sized problem, there will probably be room for tweaking these layouts for best realized performance, especially for the element fields (which might be many-dimensional, with a vector/matrix at each quadrature point).  We'll also probably want a deterministic ordering in our linear algebra data structures, since I know fill-in can be a problem with some orderings when using direct solvers.

Reply:
Yes that all sounds fine. The thing I was wondering, is for your GetNodalData function you're assuming contiguous data for a node, which is potentially true for one layout but not the other. (Unless the subview can hide/abstract that in a way that I'm not familiar with.)

Reply:
Yes - Subview abstracts it away so that it works regardless of the initial layout.  The Subview layout will generally be some weird strided thing of appropriate dimension that we don't have to worry about.  I don't pretend to know how it figures it all out. Probably magic.